### DISCUSSION TITLE
dynamic components is not picking up correctly

### DISCUSSION THREAD

**Discussion by dev02iggy** [nuxt/nuxt#16601] [General]
I'm not sure if it's a bug in Nuxt or if I spelled it wrong.
`<component :is="linkIsExternal(item.link) ? 'a' : 'NuxtLink'" :to="item.link" :href="item.link">{{item.name}}</component>`
This code snippet in Nuxt 2 worked, but in 3 it only works when the result gives "a", but when the result is "NuxtLink" it doesn't. I tried using "nuxt-link", "nuxtlink" and it doesn't work.

**Comment by sduncan-mbt** [comment]
Hi, I've been playing around with dynamic components today and had a little trouble myself. Not sure if its connected but thought it worth putting in here.

When I use the option api style I can get it to work but when trying to use a setup script the only why I can get it to work is by adding a <script> tag and importing the components there.

I've created a repo below to show you what I mean

https://github.com/sduncan-mbt/dynamic-comp

Any ideas or help much appreciated thanks.

**Comment by TheAlexLichter** [comment]
@dev02iggy For what it's worth - `NuxtLink` handles external links now too, so you don't need to switch between `a` and `NuxtLink` anymore ☺️ 

More [in the docs](https://v3.nuxtjs.org/api/components/nuxt-link)

**Comment by dev02iggy** [comment]
I understood. Thanks @manniL

**Comment by gregg-cbs** [comment]
Hi All,

Battling to get someone to look at my discussion: https://github.com/nuxt/framework/discussions/8095

In how to dynamically import components. Any help would be greatly appreciated.
